From a10a741d84b3019b720f19deb3a9622640b883bc Mon Sep 17 00:00:00 2001
From: Raymond Gauthier <jraygauthier@gmail.com>
Date: Thu, 20 Feb 2020 11:26:35 -0500
Subject: [PATCH] Device lost hang: no more src stop on device lost

No longer allow `gst_tcam_src_device_lost_callback` to call
`gst_tcam_src_stop` as `gst_tcam_src_device_lost_callback` is called
during the time `gst_tcam_src_stop` holds its lock on `self->mtx` and
this mutex is **not recursive** which lead to undefined behavior
(recursively locking the non recursive mutex).
---
 src/gstreamer-1.0/gsttcamsrc.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gstreamer-1.0/gsttcamsrc.cpp b/src/gstreamer-1.0/gsttcamsrc.cpp
index 5ee4f01..f8ed54f 100644
--- a/src/gstreamer-1.0/gsttcamsrc.cpp
+++ b/src/gstreamer-1.0/gsttcamsrc.cpp
@@ -1064,7 +1064,7 @@ static void gst_tcam_src_device_lost_callback (const struct tcam_device_info* in
 
 #endif
 
-    gst_tcam_src_stop(GST_BASE_SRC(self));
+    // gst_tcam_src_stop(GST_BASE_SRC(self));
 }
 
 
-- 
2.23.0

